bugbug/bug_features.py

```diff
@@ -475,21 +488,41 @@ def transform(self, bugs):
 
         already_rollbacked = set()
 
-        def apply_transform(bug):
-            bug_id = bug["id"]
+        def apply_transform(bug, tuples=False):
+
+            if not tuples:
+                bug_id = bug["id"]
+
+                if self.rollback and bug_id not in already_rollbacked:
+                    bug = bug_snapshot.rollback(bug, self.rollback_when)
+                    already_rollbacked.add(bug_id)
+
+            else:
+                bug1_id = bug[0]["id"]
+                bug2_id = bug[1]["id"]
 
-            if self.rollback and bug_id not in already_rollbacked:
-                bug = bug_snapshot.rollback(bug, self.rollback_when)
-                already_rollbacked.add(bug_id)
+                if self.rollback:
+                    if bug1_id not in already_rollbacked:
+                        bug[0] = bug_snapshot.rollback(bug[0], self.rollback_when)
+                        already_rollbacked.add(bug1_id)
+                    if bug2_id not in already_rollbacked:
+                        bug[1] = bug_snapshot.rollback(bug[1], self.rollback_when)
+                        already_rollbacked.add(bug2_id)
 
             data = {}
 
             for feature_extractor in self.feature_extractors:
-                res = feature_extractor(
-                    bug,
-                    reporter_experience=reporter_experience_map[bug["creator"]],
-                    author_ids=author_ids,
-                )
+                res = None
+                if isinstance(feature_extractor, single_bug_feature) and not tuples:
+
+                    res = feature_extractor(
+                        bug,
+                        reporter_experience=reporter_experience_map[bug["creator"]],
+                        author_ids=author_ids,
+                    )
+
+                elif isinstance(feature_extractor, tuple_bug_feature) and tuples:
+                    res = feature_extractor(bug)
```

Comment:
Remove `and tuples` and do `assert tuples, "Couple features can only be used for couples"` in the branch

Reply:
This way we ensure single bug models are not trying to use features for couples.

Reply:
Actually we can't do this here, as in the couples case we run both kinds of feature extractors.